src/Adapters/EventSource.Net46/Implementation/EventSourceListenerEventSource.cs

```diff
@@ -0,0 +1,33 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="EventSourceListenerEventSource.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.ApplicationInsights.EventSourceListener.Implementation
+{
+    using System.Diagnostics.Tracing;
+
+    /// <summary>
+    /// EventSource for reporting errors and warnings from the EventSourceListener telemetry module.
+    /// </summary>
+    [EventSource(Name = ProviderName)]
+    internal sealed class EventSourceListenerEventSource : EventSource
+    {
+        public const string ProviderName = "Microsoft-ApplicationInsights-Extensibility-EventSourceListener";
+        public static readonly EventSourceListenerEventSource Log = new EventSourceListenerEventSource();
+
+        private const int NoEventSourcesConfiguredEventId = 1;
+        [Event(NoEventSourcesConfiguredEventId, Level = EventLevel.Warning, Keywords = Keywords.Configuration,
```

Comment:
we report the face "application name" with every diagnostics trace to simplify IIS-based apps troubleshooting. Can you please re-use the approach